app/src/main/java/com/example/vanahel/tasksmanagerapplication/adapter/ViewPagerAdapter.java

```diff
@@ -0,0 +1,39 @@
+package com.example.vanahel.tasksmanagerapplication.adapter;
+
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentManager;
+import android.support.v4.app.FragmentPagerAdapter;
+
+import com.example.vanahel.tasksmanagerapplication.tab.AllTabFragment;
+import com.example.vanahel.tasksmanagerapplication.tab.FavoriteTabFragment;
+
+public class ViewPagerAdapter extends FragmentPagerAdapter {
+
+    int tabCount;
+
+    public ViewPagerAdapter (FragmentManager fragmentManager, int tabCount) {
+        super(fragmentManager);
+        this.tabCount= tabCount;
+    }
+
+    @Override
+    public Fragment getItem(int position) {
+        switch (position) {
+            case 0:
+                AllTabFragment allTabFragment = new AllTabFragment();
+                return allTabFragment;
+            case 1:
+                FavoriteTabFragment favoriteTabFragment = new FavoriteTabFragment();
+                return favoriteTabFragment;
+            default:
+                return null;
+        }
+    }
+
+
+
+    @Override
+    public int getCount() {
+        return tabCount;
```

Comment:
getCount() method should be managed by ViewPagerAdapter itself, not by a param from constructor. Because if a programer sets incorrect value into constructor (by mistake) the app will crash, because a ViewPager will try to instantiate a gragment with index 2 and getItem() method will return null in this situation.